addons/sale_loyalty/tests/test_program_numbers.py

```diff
@@ -1804,3 +1804,31 @@ def test_loyalty_card_tax_total(self):
         self.assertEqual(order.order_line[0].tax_id, tax_15pc_excl)
         self.assertEqual(order.order_line[1].tax_id, tax_15pc_excl)
         self.assertEqual(order.amount_total, 156.0, '140$ + 15% - 5$ = 156$')
+
+    def test_sol_free_product_description_equals_reward_description(self):
+        """
+        Ensure that when a product is added to the sale order line containing a loyalty reward (Buy X Get Y),
+        the free product's name comes from the reward's discount line product instead of the reward_product_id.
```

Comment:
```suggestion
        Ensure that if a Buy X Get Y reward is added to a sale order,
        its line description matches the reward description.
```
There's a 100 char line length limit in this scope, so the docstring can get shortened a bit.